w3ctrace/state.go

```diff
@@ -17,109 +18,116 @@ const maxKVPairs = 32
 // Length of entries that should be filtered first in case, if tracestate has more than `maxKVPairs` items
 const thresholdLen = 128
 
-// State is list of key=value pairs representing vendor-specific data in the trace context
-type State []string
+var instanaListMemberRegex = regexp.MustCompile("^\\s*" + VendorInstana + "\\s*=\\s*([^,]*)\\s*$")
 
-// ParseState parses the value of `tracestate` header. Empty list items are omitted.
-func ParseState(traceStateValue string) (State, error) {
-	states := filterEmptyItems(strings.Split(traceStateValue, ","))
-	if len(states) < maxKVPairs {
-		return states, nil
-	}
+// State is list of key=value pairs representing vendor-specific data in the trace context
+type State struct {
+	// The key-value pairs of other vendors. The "in" list member, if present, will be stored separately
+	// (Note: list member is the term the W3C trace context specfication uses for the key-value pairs.)
+	listMembers []string
 
-	itemsToFilter := len(states) - maxKVPairs
-	filteredStates := states[:0]
+	// The value for the Instana-specific list member ("in=...").
+	instanaTraceStateValue string
+}
 
-	i := 0
-	for ; itemsToFilter > 0 && i < len(states); i++ {
+// makeState creates a new State with the given values
+func MakeState(listMembers []string, instanaTraceStateValue string) State {
+	return State{
+		listMembers:            listMembers,
+		instanaTraceStateValue: instanaTraceStateValue,
+	}
+}
 
-		if len(states[i]) > thresholdLen {
-			itemsToFilter--
-			continue
+// ParseState parses the value of `tracestate` header. Empty list items are omitted.
+func ParseState(traceStateValue string) (State, error) {
+	listMembers := filterEmptyItems(strings.Split(traceStateValue, ","))
+
+	// Look for the Instana list member first, before discarding any list members due to length restrictions.
+	instanaTraceStateValue := ""
```

Comment:
Optional, but mostly for you knowledge (if you don't know it already), you can declare a var without giving it a value, because in Go, every var has always its "zero" value by default. eg: numbers is 0, strings is "" and so on :) 

```go
var instanaTraceStateValue string
```

Reply:
Done.